sessiontxn/isolation/base.go

```diff
@@ -175,20 +176,43 @@ func (p *baseTxnContextProvider) activateTxn() (kv.Transaction, error) {
 		return nil, err
 	}
 
-	txn, err := p.sctx.Txn(true)
+	txnFuture := p.sctx.GetPreparedTxnFuture()
+	if txnFuture == nil {
+		return nil, errors.AddStack(kv.ErrInvalidTxn)
+	}
+
+	txn, err := txnFuture.Wait(p.ctx, p.sctx)
 	if err != nil {
 		return nil, err
 	}
 
 	sessVars := p.sctx.GetSessionVars()
 	sessVars.TxnCtx.StartTS = txn.StartTS()
 
+	if !sessVars.IsAutocommit() && sessVars.SnapshotTS == 0 {
```

Comment:
When `EnterNewTxnDefault` and `autocommit=0` it will also enter an explicit txn? We should keep the behavior the same with the old implement. And we need to add more tests for this case.

Reply:
I have modified it.